### PR TITLE
add dstu2 order object

### PR DESCRIPTION
**Pull request by ItayGoren**
@nazrulworld 
Before I add the tests, please take a look if I done that correctly:
- Did I define the fields correctly with the attributes?
- I understood I need to add them to `fhirtypes.py`, is there another place I need to add them?

### CODE REVIEW

**Comment by nazrulworld** [comment]
Also need to add an entry at https://github.com/nazrulworld/fhir.resources/blob/master/fhir/resources/DSTU2/fhirtypesvalidators.py

**Comment by ItayGoren** [comment]
@nazrulworld done and added tests
review would be appreciated

File: fhir/resources/DSTU2/order.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    detail: ListType[fhirtypes.ReferenceType] = Field(
+        ...,
+        alias="detail",
+        title="detail",
+        description="What action is being ordered.",
+        # if property is element of this resource.
+        element_property=True,
+    )
+
```
Comment (nazrulworld) [inline fhir/resources/DSTU2/order.py:131]: root validator is missing, please have a look at the example here https://github.com/nazrulworld/fhir.resources/blob/master/fhir/resources/DSTU2/visionprescription.py#L108

**Review by nazrulworld — COMMENTED** [review]
1. Update AUTHORS.rst
2. Check lint `make lint`(optional)

**Comment by nazrulworld** [comment]
@ItayGoren About "validate on of many" 

First of all,  all classes do not contain those special fields, secondly, if you look at here, field names are varied, in different classes. 
```
one_of_many_fields = {
            "reason": ["reasonCodeableConcept", "reasonReference"],
}
```

Thirdly, yes it might possible to make a general root validator, which would capable to scan all fields and do validate on demand, but that will make extra overhead for those classes don't have one of many fields.

**Comment by nazrulworld** [comment]
LGTM

**Pull request merged**